rq/job.py

```diff
@@ -591,8 +624,7 @@ def delete(self, pipeline=None, remove_from_queue=True,
         if delete_dependents:
             self.delete_dependents(pipeline=pipeline)
 
-        connection.delete(self.key)
-        connection.delete(self.dependents_key)
+        connection.delete(self.key, self.dependents_key, self.dependencies_key)
```

Comment:
`dependencies_key`, along with `dependents_key` should be expired in [job.cleanup()](https://github.com/rq/rq/blob/master/rq/job.py#L655)

Reply:
:+1: 71a427279f90c5ab566ba741467ad2413994bc75